Source/Core/VideoBackends/Vulkan/VKScheduler.h

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <cstddef>
+#include <queue>
+
+#include "CommandBufferManager.h"
+#include "Common/Align.h"
+
+namespace Vulkan
+{
+class Scheduler
+{
+  class Command
+  {
+  public:
+    virtual ~Command() = default;
+
+    virtual void Execute(CommandBufferManager* cmdbuf) = 0;
+
+    Command* GetNext() const { return next; }
```

Comment:
Instead of this GetNext/SetNext have you considered having a virtual GetSize which returns sizeof(*this) in TypedCommand? IMO this is a more natural design than the linked list, but I'm curious what your thoughts are on this. I haven't deeply thought about the pros/cons of either approach other than "I would implemented this with a GetSize instead".

Reply:
This originally comes from DXVK. To be perfectly honest, I didn't really put a lot of thought into changing anything because the code is battle tested and works great with some of the latest and greatest AAA games in DXVK that naturally push it much harder than Dolphin ever will.